### PR TITLE
 find first non-repeating character in a string in java  #569

### PR DESCRIPTION
**Pull request by ASLManasa**
#569 Code using in java @tarun26091999  sir can you please merge it.

### CODE REVIEW

**Comment by ASLManasa** [comment]
@raksha009 @shreyanspoddar  Sir Can you please review this code. Thank you. @tarun26091999

**Comment by ASLManasa** [comment]
@raksha009 @shreyanspoddar  Sir Can you please review this code. Thank you. @tarun26091999

**Review by raksha009 — CHANGES_REQUESTED** [review]
Plz link yr code with the appropriate file. Go through the guidelines and pinned issues.

File: string/first_non_repeating_charachters [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/* The program will find the first non-repeating character in the string if it is present in the string */
```
Comment (raksha009) [inline string/first_non_repeating_charachters:1]:
```suggestion
/* 
The program will find the first non-repeating character in the string if it is present in the string 
*/
```
Plz add a bit more detailed description here

File: string/first_non_repeating_charachters [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Scanner;
+
+public class first_non_repeating_charachters {
```
Comment (raksha009) [inline string/first_non_repeating_charachters:6]:
```suggestion
public class first_non_repeating_characters {
```
Plz correct the spelling of character in file name too.

File: string/first_non_repeating_charachters [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public static void non_repeating(String s) {
```
Comment (raksha009) [inline string/first_non_repeating_charachters:7]: Add a newline above line 7

File: string/first_non_repeating_charachters [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        int frequency_of_char[] = new int[256];
+        //creating an array that calculates the frequency of all the charachter in the string
+        for (int i = 0; i < s.length(); i++)
+            frequency_of_char[(int) s.charAt(i)]++;
+        //calculating the frequency of the char
```
Comment (raksha009) [inline string/first_non_repeating_charachters:12]: Add a newline above line 12

File: string/first_non_repeating_charachters [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        for (int j = 0; j < 256; j++) {
+            if (frequency_of_char[j] == 1)
+            //if the frequency of a char is 1 then print and stop the loop.
+            {
+                System.out.println("The first non repeating charachter is  : " + (char) j);
+                break;
+            }
+        }
+
```
Comment (raksha009) [inline string/first_non_repeating_charachters:21]: Remove like 21,22

File: string/first_non_repeating_charachters [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+public class first_non_repeating_charachters {
+    public static void non_repeating(String s) {
+        int frequency_of_char[] = new int[256];
+        //creating an array that calculates the frequency of all the charachter in the string
+        for (int i = 0; i < s.length(); i++)
+            frequency_of_char[(int) s.charAt(i)]++;
+        //calculating the frequency of the char
+        for (int j = 0; j < 256; j++) {
+            if (frequency_of_char[j] == 1)
+            //if the frequency of a char is 1 then print and stop the loop.
```
Comment (raksha009) [inline string/first_non_repeating_charachters:15]: add this comment above line 14

File: string/first_non_repeating_charachters [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            }
+        }
+
+
+    }
+
+    public static void main(String[] args) {
+        System.out.println("Enter the String: ");
+        Scanner sc = new Scanner(System.in);
+        //creating a scanner object
```
Comment (raksha009) [inline string/first_non_repeating_charachters:28]: Remove this comment

File: string/first_non_repeating_charachters [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        String s = sc.next().toLowerCase(Locale.ROOT).trim();
+        //taking a string input and converting into lower case and removing trailing and leading spaces
```
Comment (raksha009) [inline string/first_non_repeating_charachters:30]: Add a comment first above the line describing what the line is doing and then write that line

File: string/first_non_repeating_charachters [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        non_repeating(s);
+        //calling the function and passing string s through it
+    }
+}
+/*
+Test cases:
+1st testcase
```
Comment (raksha009) [inline string/first_non_repeating_charachters:37]: Do not write 1st testcase and so on

File: string/first_non_repeating_charachters [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+input: madam
```
Comment (raksha009) [inline string/first_non_repeating_charachters:38]:
```suggestion
Input: madam
```
Similarly O capital for Output

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by ASLManasa** [comment]
@raksha009  Mam I have made the changes If you check once I will make another pr . Thank you.

**Comment by raksha009** [comment]
Plz link yr code to the folder. Go through the guidlines and pinned issues.

**Pull request closed without merging**